doc/sphinx/language/core/assumptions.rst

```diff
@@ -71,7 +71,21 @@ the list of binders, it is expanded to a let-in definition (see
 Section :ref:`let-in`).
 
 The expression :n:`forall @ident : @type, @term` denotes the
-*product* of the variable :n:`@ident` of type :n:`@type`, over the term :n:`@term`.
+:gdef:`product type <product>` (or *product*) of the variable :n:`@ident` of type :n:`@type`,
+over the term :n:`@term`.  If :n:`@ident` is used in :n:`@term`, then we say
+the expression is a :gdef:`dependent product`.
```

Comment:
There is a redundancy with a paragraph later in this section. (Maybe the two can be merged.)

Reply:
I dropped the first sentence and moved it, but I don't fully understand the content.